synapse/storage/databases/main/user_directory.py

```diff
@@ -506,13 +514,21 @@ async def add_users_in_public_rooms(
             user_ids
         """
 
-        await self.db_pool.simple_upsert_many(
-            table="users_in_public_rooms",
-            key_names=["user_id", "room_id"],
-            key_values=[(user_id, room_id) for user_id in user_ids],
-            value_names=(),
-            value_values=None,
-            desc="add_users_in_public_rooms",
+        def _add_users_in_public_rooms_txn(txn):
```

Comment:
(Not related to this PR, but can you fix the docstring above this for `add_users_in_public_rooms`? It refers to the wrong table.)